Add code formatter

NoZiL opened the pull request.

GaelTouquet commented:
The conflicts are too complicated to handle by hand with all the changes that have been added to the code. Can I just apply the code formatter to the new version of the code? if yes how?
Also noticed that the requirements were updated, anything else has changed?

NoZiL committed:
Add code formatter

NoZiL committed:
Apply format (>Make format)

NoZiL commented:
I've split the commit into
1. containing the code for formatting
2. applying the new format rules with the command `Make format` and rebased on the upstream master to have the latest version
